### ISSUE TITLE
Different results between 17.4 and 17.5+

### ISSUE THREAD

**Issue by RussKie** [microsoft/codecoverage#5]
I have two identical builds that differ by the version of Microsoft.CodeCoverage. 

The build that uses 17.4.1 reports higher metrics, and builds that use 17.5.x or 17.6.y report lower metrics:

17.4.1 ([build](https://dev.azure.com/dnceng/internal/_build/results?buildId=2119121&view=codecoverage-tab))
![image](https://user-images.githubusercontent.com/4403806/220304592-86dc50ab-f639-4c9f-b3e8-e33195ad86b0.png)

17.6.0-preview-20230220-06 ([build](https://dev.azure.com/dnceng/internal/_build/results?buildId=2119235&view=codecoverage-tab))
![image](https://user-images.githubusercontent.com/4403806/220304694-e2a26e41-748c-4631-afc9-213460dd30a3.png)

Looking at the reports side-by-side I can't see any difference:
![image](https://user-images.githubusercontent.com/4403806/220303492-8f2c7f47-5a8e-4a46-b943-4225ca2df8de.png)

What am I missing?

Please note, the source repo is not currently public, please ping me in Teams, if you need more details.

/cc: @jakubch1  @tekian @joperezr

**Comment by jakubch1** [comment]
@fhnaseer please investigate.

**Comment by fhnaseer** [comment]
There was a [bug](https://github.com/microsoft/vstest/issues/3940) which we fixed in 17.6. So 17.6 shows correct numbers.

17.4 Report:
```
<line number="117" hits="24" branch="true" condition-coverage="100% (4/4)" />
<line number="118" hits="24" branch="true" condition-coverage="100% (4/4)" />
<line number="119" hits="24" branch="true" condition-coverage="100% (4/4)" />
```
17.6 Report:
```
<line number="117" hits="24" branch="true" condition-coverage="100% (2/2)" />
<line number="118" hits="24" branch="true" condition-coverage="100% (2/2)" />
<line number="119" hits="24" branch="true" condition-coverage="100% (2/2)" />
```
Code:
![image](https://user-images.githubusercontent.com/1503766/220642257-eea6b124-0098-4538-9aaf-82ade01c9e45.png)

For all these lines, there are only 2 conditions and hence correct value is (2/2) rather than (4/4).

**Comment by RussKie** [comment]
Thank you for the confirmation.
